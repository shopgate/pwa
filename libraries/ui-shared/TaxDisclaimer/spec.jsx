import React from 'react';
import { shallow } from 'enzyme';
import I18n from '@shopgate/pwa-common/components/I18n';

jest.mock('@shopgate/engage/core/hooks/useWidgetSettings', () => ({
  useWidgetSettings: jest.fn().mockReturnValue({
    show: null,
    hint: '*',
    text: null,
  }),
}));

describe('<TaxDisclaimer />', () => {
  afterEach(() => {
    jest.resetModules();
  });

  it('should display the component', () => {
    jest.mock('@shopgate/pwa-common-commerce/market/helpers/showTaxDisclaimer', () => true);
    // eslint-disable-next-line global-require
    const TaxDisclaimer = require('./index').default;
    const wrapper = shallow(<TaxDisclaimer />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(I18n.Text).exists()).toBe(true);
  });

  it('should display null', () => {
    jest.mock('@shopgate/pwa-common-commerce/market/helpers/showTaxDisclaimer', () => false);
    // eslint-disable-next-line global-require
    const TaxDisclaimer = require('./index').default;
    const wrapper = shallow(<TaxDisclaimer />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(I18n.Text).exists()).toBe(false);
  });
});
