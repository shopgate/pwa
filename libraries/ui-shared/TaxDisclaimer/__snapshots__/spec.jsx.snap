// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaxDisclaimer /> should display null 1`] = `
<SurroundPortals
  portalName="product.tax-disclaimer"
  portalProps={
    Object {
      "showTaxDisclaimer": false,
    }
  }
/>
`;

exports[`<TaxDisclaimer /> should display the component 1`] = `
<SurroundPortals
  portalName="product.tax-disclaimer"
  portalProps={
    Object {
      "showTaxDisclaimer": true,
    }
  }
>
  <div
    aria-hidden={true}
    className="ui-shared__tax-disclaimer"
    data-test-id="taxDisclaimer"
  >
    <Translate
      className="css-fazkyv"
      params={Object {}}
      role={null}
      string="product.tax_disclaimer"
    />
  </div>
</SurroundPortals>
`;
